common/lib/xmodule/xmodule/modulestore/tests/test_split_mongo_mongo_connection.py

```diff
@@ -0,0 +1,21 @@
+""" Test the behavior of split_mongo/MongoConnection """
+import unittest
+from mock import patch
+from xmodule.modulestore.split_mongo.mongo_connection import MongoConnection
+from xmodule.exceptions import HeartbeatFailure
+
+
+class TestHeartbeatFailureException(unittest.TestCase):
+    """ Test that a heartbeat failure is thrown at the appropriate times """
+    @patch('pymongo.MongoClient')
+    @patch('pymongo.database.Database')
+    def test_heartbeat_raises_exception_when_connection_alive_is_false(self, *calls):
+        # pylint: disable=W0613
+        with patch('mongodb_proxy.MongoProxy') as mock_proxy:
+            mock_proxy.return_value.alive.return_value = False
+            useless_conn = MongoConnection('useless', 'useless', 'useless')
+
+            with self.assertRaises(HeartbeatFailure) as context:
+                useless_conn.heartbeat()
+
+            self.assertTrue(context.exception.__class__ == HeartbeatFailure)
```

Comment:
Nit: I don't think you actually need this last assert because line 18 covers verifying that a HeartbeatFailure was raised.